CivetWeb Enable WebSocket

**Is your enhancement proposal related to a problem? Please describe.**
Compiling **CivetWeb** for `nucleo_f767zi` board with `WebSocket` support is not possible.

Steps to reproduce are:
1. In `zephyrproject/modules/lib/civetweb/CMakeLists.txt` enable the option `CIVETWEB_ENABLE_WEBSOCKETS`
2. Compile the project as follows: `west build -p auto -b "nucleo_f767zi" "samples/net/sockets/civetweb"`
3. Following error should occur:
```
<some_your_path>/zephyrproject/modules/lib/civetweb/src/civetweb.c: In function 'websocket_client_thread':
<some_your_path>/zephyrproject/modules/lib/civetweb/src/civetweb.c:17311:19: error: storage size of 'sa' isn't known 17311 |  struct sigaction sa;
```

It seems to be, that there is no `signal.h` implementation for any `stm32`!

**Describe the solution you'd like**
Clear and proper project compilation with `CIVETWEB_ENABLE_WEBSOCKETS` option.

The file `<some_your_path>/zephyrproject/modules/hal/esp-idf/components/newlib/include/sys/signal.h` seems to have necessary structure. 

**Additional context**
According to above info I have following questions:
* Does anybody already up to integration of WebSockets within CivetWeb to Zephyr and/or stm32?
* Is it possible and/or necessary to adapt `<some_your_path>/zephyrproject/modules/hal/esp-idf/components/newlib/include/sys/signal.h` for stm32?

--- a/west.yml
+++ b/west.yml
@@ -41,7 +41,7 @@ manifest:
       revision: da9a2df574094f52d87a03f6393928bdc7dce17c
       path: tools/ci-tools
     - name: civetweb
-      revision: 99129c5efc907ea613c4b73ccff07581feb58a7a
+      revision: e6903b80c09d17cd1a8bb32e40080005558aad29
       path: modules/lib/civetweb
     - name: esp-idf
       revision: 6835bfc741bf15e98fb7971293913f770df6081f

